### PR TITLE
Added design principles to introduction

### PR DESCRIPTION
**Pull request by adrianhall**

### CODE REVIEW

File: docs/design/Introduction.mdk [hunk 1]
```diff
@@ -1,20 +1,56 @@
...
+### Idiomatic
+
+* The SDK should follow the general design guidelines and conventions for the target language.  It should feel natural to a developer in the target language.
+* We embrace the ecosystem with its strengths and its flaws.
+* We work with the ecosystem to improve it for all developers.
+
+### Consistent
+
+* Client libraries should be consistent within the language, consistent between all target languages, and consistent with the service.  In cases of conflict, consistency within the language is the highest priority and consistency with the service is the lowest priority.
+* Service-agnostic concepts, such as logging, HTTP connection management, and error handling, should be consistent.  The developer should not have to relearn service-agnostic concepts as they move between client libraries.
```
Comment (JeffreyRichter) [inline docs/design/Introduction.mdk]: I think HTTP communication is better than "HTTP connection management" as HTTP is generally considered a connectionless protocol.

File: docs/design/Introduction.mdk [hunk 2]
```diff
@@ -1,20 +1,56 @@
...
+* Consistency of terminology between the client library and the service is a good thing that aids in diagnosability.  
+* All differences must have a good (articulated) reason for existing, rooted in idiomatic usage rather than whim.
```
Comment (JeffreyRichter) [inline docs/design/Introduction.mdk]: All difference in what? I think you need a noun here.

File: docs/design/Introduction.mdk [hunk 3]
```diff
@@ -1,20 +1,56 @@
...
+* The Azure SDK for each target language feels like a single product developed by a single team.
+* There should be feature parity across target languages.  This is more important than feature parity with the service.
+
+### Approachable
+
+* We are experts in the supported technologies so our customers, the developers, don't have to be.
+* Developers should find great documentation (hero tutorial, how to articles, samples, and API documentation) that makes it easy to be successful with the Azure service.
+* Getting off the ground should be easy through the use of predictable defaults that implement best practices.  Think about progressive concept disclosure.
+* The SDK should be easily acquired through the most normal mechanisms in the target language and ecosyste.
```
Comment (JeffreyRichter) [inline docs/design/Introduction.mdk]: ecosystem <-- typo

File: docs/design/Introduction.mdk [hunk 4]
```diff
@@ -1,20 +1,56 @@
...
+* Developers can be overwhelmed when learning new service concepts.  The core use cases should be discovered
```
Comment (JeffreyRichter) [inline docs/design/Introduction.mdk]: The core use cases should be easily discoverable.

File: docs/design/Introduction.mdk (hunk 1 above)
Comment (bterlson) [reply docs/design/Introduction.mdk]: HTTP as of 1.1 is not very connectionless, and 2.0/3.0 are even less so :) That said, maybe "HTTP session" is the word we're looking for here.

File: docs/design/Introduction.mdk [hunk 5]
```diff
@@ -1,20 +1,56 @@
...
+
+### Diagnosable
+
+* The developer should be able to understand what is going on.
+* It should be discoverable when and under what circumstances a network call is made.
+* It is discoverable what the defaults are and what their intent is.
+* Logging, tracing, and exception handling are fundamental and should be thoughtful.
+* Error messages should be concise, correlated with the service, actionable, and human readable.
```
Comment (JeffreyRichter) [inline docs/design/Introduction.mdk]: Maybe add "and ideally leading the customer to a useful action that they can take"

File: docs/design/Introduction.mdk (hunk 2 above)
Comment (bterlson) [reply docs/design/Introduction.mdk]: I read this as "all differences (between the client library and the service)", fwiw

I'll also observe that this principle argues for exposing block/page/append blobs?

File: docs/design/Introduction.mdk [hunk 6]
```diff
@@ -1,20 +1,56 @@
...
-- **Client Library**. This refers to a library that customers/developers use to ease working with an Azure service. There is often one client library per service and per programming language. Sometimes a single client library will contain the ability to connect to multiple services. 
-
 - **Package**. This refers to a client library after it has been packaged for distribution for customer-developers to consume. Examples are:
    - A Nuget package for a .NET client library
    - A Maven package for a Java library
    - An NPM package for a JavaScript library
-   - A PiPI package for a Python library
-   - A GitHub repositiory for a Go library (NOTE: Go is not a supported language yet)
\ No newline at end of file
+   - A Pypi package for a Python library
```
Comment (johanste) [inline docs/design/Introduction.mdk]: A python wheel for a python library.

File: docs/design/Introduction.mdk (hunk 2 above)
Comment (adrianhall) [reply docs/design/Introduction.mdk]: It's also between target languages... I think the between service and client library is another good one, though, so I'll add it.

File: docs/design/Introduction.mdk (hunk 5 above)
Comment (adrianhall) [reply docs/design/Introduction.mdk]: Agreed,  Fixed.

File: docs/design/Introduction.mdk (hunk 4 above)
Comment (adrianhall) [reply docs/design/Introduction.mdk]: Thanks - fixed.

File: docs/design/Introduction.mdk (hunk 3 above)
Comment (adrianhall) [reply docs/design/Introduction.mdk]: Thanks - fixed.

File: docs/design/Introduction.mdk (hunk 1 above)
Comment (adrianhall) [reply docs/design/Introduction.mdk]: I used HTTP communication since it encompasses all these ideas.

**Pull request merged**